templates/_composition/_shared/Page.Camera.AddResource/Controls/CameraControl.xaml.cs

```diff
@@ -65,24 +72,107 @@ public bool IsInitialized
+
+            Loaded += OnLoaded;
+            Unloaded += OnUnloaded;
+        }
+
+        public async Task InitializeCameraAsync()
         {
             try
             {
-                await InitializeCameraAsync();
+                if (_mediaCapture == null)
+                {
+                    _mediaCapture = new MediaCapture();
+                    _mediaCapture.Failed += MediaCapture_Failed;
+
+                    _cameraDevices = await DeviceInformation.FindAllAsync(DeviceClass.VideoCapture);
+                    if (_cameraDevices == null || !_cameraDevices.Any())
+                    {
+                        throw new NotSupportedException();
+                    }
+
+                    var device = _cameraDevices.FirstOrDefault(camera => camera.EnclosureLocation?.Panel == Panel);
+
+                    var cameraId = device?.Id ?? _cameraDevices.First().Id;
+
+                    await _mediaCapture.InitializeAsync(new MediaCaptureInitializationSettings { VideoDeviceId = cameraId });
+
+                    if (Panel == Panel.Back)
+                    {
+                        _mediaCapture.SetRecordRotation(VideoRotation.Clockwise90Degrees);
+                        _mediaCapture.SetPreviewRotation(VideoRotation.Clockwise90Degrees);
+                        _mirroringPreview = false;
+                    }
+                    else
+                    {
+                        _mirroringPreview = true;
+                    }
+
+                    IsInitialized = true;
+                    CanSwitch = _cameraDevices?.Count > 1;
+                    RegisterOrientationEventHandlers();
+                    await StartPreviewAsync();
+                }
             }
             catch (UnauthorizedAccessException ex)
             {
                 throw new UnauthorizedAccessException("Camera_Exception_UnauthorizedAccess".GetLocalized(), ex);
             }
             catch (NotSupportedException ex)
             {
-                throw new NotSupportedException("Camera_Exception_NotSuppored".GetLocalized(), ex);
+                throw new NotSupportedException("Camera_Exception_NotSupported".GetLocalized(), ex);
+            }
+        }
+
+        public async Task CleanupCameraAsync()
+        {
+            if (IsInitialized)
+            {
+                if (_isPreviewing)
+                {
+                    await StopPreviewAsync();
+                }
+
+                UnregisterOrientationEventHandlers();
+                IsInitialized = false;
+            }
+
+            if (_mediaCapture != null)
+            {
+                _mediaCapture.Failed -= MediaCapture_Failed;
+                _mediaCapture.Dispose();
+                _mediaCapture = null;
             }
         }
 
-        public async Task<string> TakePhotoAsync()
+        public async Task<string> TakePhoto()
         {
+            if (_capturing)
```

Comment:
You are never setting _capturing back to false.

Reply:
You're right, it's already fixed.